tests/api/utils/singletons/test_k8s_utils.py

```diff
@@ -58,8 +68,68 @@ def test_get_logger_pods_label_selector(
     if extra_selector:
         selector += f",{extra_selector}"
 
-    k8s_helper = server.api.utils.singletons.k8s.K8sHelper(namespace, silent=True)
     k8s_helper.list_pods = unittest.mock.MagicMock()
 
     k8s_helper.get_logger_pods(project, uid, run_type)
-    k8s_helper.list_pods.assert_called_once_with(namespace, selector=selector)
+    k8s_helper.list_pods.assert_called_once_with(
+        k8s_helper.namespace, selector=selector
+    )
+
+
+@pytest.mark.parametrize(
+    "k8s_secret_data,secrets_data",
+    [
+        ({"key1": "value1", "key2": "value2"}, []),
+        ({"key1": "value1", "key2": "value2"}, None),
+        (None, ["key1"]),
+        ({}, ["key1"]),
+    ],
+)
```

Comment:
Maybe add a case where they both exists but the keys don't overlap? just to make sure it's not breaking anything